modules/charon-core/src/main/java/org/wso2/charon3/core/objects/Role.java

```diff
@@ -0,0 +1,352 @@
+            MultiValuedAttribute users = (MultiValuedAttribute) this.attributeList
+                    .get(SCIMConstants.RoleSchemaConstants.USERS);
+            List<Attribute> values = users.getAttributeValues();
+            if (values != null) {
+                List<Attribute> subValuesList = users.getAttributeValues();
+                for (Attribute subValue : subValuesList) {
+                    ComplexAttribute complexAttribute = (ComplexAttribute) subValue;
+                    Map<String, Attribute> subAttributesList = complexAttribute.getSubAttributesList();
+                    if (subAttributesList != null && subAttributesList
+                            .containsKey(SCIMConstants.CommonSchemaConstants.DISPLAY)) {
+                        displayNames.add((String) ((SimpleAttribute) (subAttributesList
+                                .get(SCIMConstants.CommonSchemaConstants.DISPLAY))).getValue());
+                    }
+
+                }
+                return displayNames;
+            }
+        }
+
+        return displayNames;
+    }
+
+    /**
+     * Set a user to the role, where user will have three default attributes such as name, value and $ref.
+     *
+     * @param user User object.
+     * @throws BadRequestException BadRequestException.
+     * @throws CharonException     CharonException.
+     */
+    public void setUser(User user) throws BadRequestException, CharonException {
+
+        if (this.isAttributeExist(SCIMConstants.RoleSchemaConstants.USERS)) {
+            MultiValuedAttribute users = (MultiValuedAttribute) this.attributeList
+                    .get(SCIMConstants.RoleSchemaConstants.USERS);
+            ComplexAttribute complexAttribute = processUser(user);
+            users.setAttributeValue(complexAttribute);
+        } else {
+            MultiValuedAttribute users = new MultiValuedAttribute(SCIMConstants.RoleSchemaConstants.USERS);
+            DefaultAttributeFactory.createAttribute(SCIMSchemaDefinitions.SCIMRoleSchemaDefinition.USERS, users);
+            ComplexAttribute complexAttribute = processUser(user);
+            users.setAttributeValue(complexAttribute);
+            this.setAttribute(users);
+        }
+    }
+
+    /**
+     * Create user attribute with three default attributes such as name, value and $ref.
+     *
+     * @param user User object.
+     * @return User object as a complex attribute.
+     * @throws BadRequestException BadRequestException.
+     * @throws CharonException     CharonException.
+     */
+    private ComplexAttribute processUser(User user) throws BadRequestException, CharonException {
+
+        String userId = user.getId();
+        String userName = user.getUserName();
+        String reference = user.getLocation();
+
+        ComplexAttribute complexAttribute = new ComplexAttribute();
+        complexAttribute.setName(SCIMConstants.RoleSchemaConstants.USERS + "_" + userId + SCIMConstants.DEFAULT);
+
+        if (userId != null) {
```

Comment:
shall we use `StringUtils.isNotBlank` ?

Reply:
StringUtils is not used by the charon repo. Therefore thought of not adding the dependency as it's not really necessary for this particular case.